hnn_core/basket.py

```diff
@@ -89,45 +64,49 @@ def parreceive(self, gid, gid_dict, pos_dict, p_ext):
         for gid_src, p_src, pos in zip(gid_dict['common'],
                                        p_ext, pos_dict['common']):
             # check if AMPA params are defined in the p_src
-            if 'L2Basket_ampa' in p_src.keys():
+            if f'{self.name}_ampa' in p_src.keys():
                 # create an nc_dict
-                nc_dict_ampa = {
+                nc_dict = {
                     'pos_src': pos,
-                    'A_weight': p_src['L2Basket_ampa'][0],
-                    'A_delay': p_src['L2Basket_ampa'][1],
+                    'A_weight': p_src[f'{self.name}_ampa'][0],
+                    'A_delay': p_src[f'{self.name}_ampa'][1],
                     'lamtha': p_src['lamtha'],
                     'threshold': p_src['threshold'],
                     'type_src': 'ext'
                 }
 
-                # AMPA synapse
-                self.ncfrom_common.append(self.parconnect_from_src(
-                    gid_src, nc_dict_ampa, self.soma_ampa))
+                self._connect_feed_at_loc(
+                    feed_loc='proximal', receptor='ampa',
+                    gid_src=gid_src, nc_dict=nc_dict,
+                    nc_list=self.ncfrom_common)
 
             # Check if NMDA params are defined in p_src
-            if 'L2Basket_nmda' in p_src.keys():
-                nc_dict_nmda = {
+            if f'{self.name}_nmda' in p_src.keys():
```

Comment:
As in here (reference to https://github.com/jonescompneurolab/hnn-core/pull/146#discussion_r479299196).